packages/opentelemetry-api/src/trace/spancontext-utils.ts

```diff
@@ -13,24 +13,34 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+import { SpanContext } from './span_context';
+import { TraceFlags } from './trace_flags';
 
-import { SpanContext, TraceFlags } from '@opentelemetry/api';
-
-export const INVALID_SPANID = '0';
-export const INVALID_TRACEID = '0';
+const VALID_TRACEID_REGEX = /^([0-9a-f]{16}){1,2}$/i;
+const VALID_SPANID_REGEX = /^[0-9a-f]{16}$/i;
+const INVALID_ID_REGEX = /^0+$/i;
+export const INVALID_SPANID = '0000000000000000';
+export const INVALID_TRACEID = '00000000000000000000000000000000';
 export const INVALID_SPAN_CONTEXT: SpanContext = {
   traceId: INVALID_TRACEID,
   spanId: INVALID_SPANID,
   traceFlags: TraceFlags.NONE,
 };
 
+export function isValidTraceId(traceId: string): boolean {
+  return VALID_TRACEID_REGEX.test(traceId) && !INVALID_ID_REGEX.test(traceId);
```

Comment:
I suspect that `=== INVALID_TRACEID` would be much faster than using a regex test for the second check and we already know the length is correct because of the first regex test.

Reply:
true that